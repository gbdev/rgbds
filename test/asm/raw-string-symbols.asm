opt Wno-unmapped-char

def hello equs "world"
def name equs "hello"
println "{name}"
println #name
assert !strcmp(strsub(#name, 1, 4), "hell")
assert strlen(#hello) == charlen(#hello)
assert strlen("{hello}") == 5

def multi equs """the	quick
brown	fox"""
println #multi

def char equs "A"
def n = #char
println n
def n = (#char)
println n
def n = 1 + #char
println n
assert #char == $41

def fmt equs "%s %s %d"
println strfmt(#fmt, #name, #hello, (#char))

purge #name
assert !def(name) && !def(#name) && def(hello)

section "test", rom0
#label:
db #hello
dw #hello
dw BANK(#label), #label
