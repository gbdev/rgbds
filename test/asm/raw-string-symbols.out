hello
hello
the	quick
brown	fox
$41
$41
$42
hello world 65
